CacheMeIfYouCan/CacheMeIfYouCan/CacheTypes/FileSystemDataCache.swift

```diff
@@ -8,27 +8,46 @@
 
 import Foundation
 
+/// Stores any type which can be converted to raw `Data` on the
+/// device's local filesystem.
 open class FileSystemDataCache<T: DataConvertible>: Cache {
     
     public typealias StoredType = T
     
-    let folderURL: URL
+    let directoryURL: URL
     
     public lazy var localQueue = FileSystemDataCache.defaultQueue
     
-    public init(folderURL: URL) {
-        self.folderURL = folderURL
+    
+    /// Designated initializer.
+    ///
+    /// - Parameters:
+    ///   - rootDirectory: The root directory where the sub-directory containing this cache should be created.
+    ///   - subdirectoryName: A non-empty name for the sub-directory of the root to store items in for this cache. NOTE: An empty string will cause this initializer to fail.
+    public init?(rootDirectory: FileSystemPathHelper.UserDirectory,
+                 subdirectoryName: String) {
+        guard !subdirectoryName.isEmpty else {
+            if !TestHelper.isTesting {
+                // Tell the developer they shouldn't be doing this during development.
+                assertionFailure("You need to pass in a sub-directory name!")
```

Comment:
Minor: Could `subdirectoryName` default to a value rather than having to be specified? The requirement of `subDirectoryName` is to avoid the issue of accidentally deleting files when clearing the cache, so it is in essence, an implementation detail of this library. By allowing `subDirectoryName` to be user-set you are leaking an abstraction.

Reply:
Ooh, interesting point 🤔might have it default to the class name - otherwise if you set up different caches it'd all use the same folder, and you'd run into the same issue of accidentally deleting shit you didn't mean to.